install/helm/agones/templates/service/allocation.yaml

```diff
@@ -72,6 +72,9 @@ spec:
       protocol: TCP
 {{- end }}
   type: {{ .Values.agones.allocator.service.serviceType }}
+{{- if and (eq .Values.agones.allocator.service.serviceType "ClusterIP") (.Values.agones.allocator.service.headless) }}
```

Comment:
Rather than a special case for "headless", any reason not to do a `agones.allocator.service.clusterIP` - which can be set to whatever you want, and provides greater flexibility?

Reply:
Thank you for your comment!
Surely. `clusterIP` field which users can set any value has greater flexibility and can also be used in cases other than headless.

I changed my commit while referring to `loadBalanceIP`.
`{{- $useLoadBalancerIP := and (ne .Values.agones.allocator.service.loadBalancerIP "") (eq .Values.agones.allocator.service.serviceType "LoadBalancer") }}`